proposals/11385-literal-input-parameters/README.md

```diff
@@ -0,0 +1,131 @@
+# KEP-11385: Literal Input Parameters
+- [Summary](#summary)
+- [Motivation](#motivation)
+  - [Goals](#goals)
+  - [Non-Goals](#non-goals)
+- [Proposal](#proposal)
+  - [User stories](#user-stories)
+  - [Design Details](#design-details)
+  - [Test Plan](#test-plan)
+- [Implementation History](#implementation-history)
+- [Drawbacks](#drawbacks)
+
+## Summary
+It is not currently possible to limit pipeline input to a specified set of values. Often pipeline authors want to restrict input in order to reduce unintended behavior and pipeline failure. This KEP proposes extending the valid pipeline input types to include Python Literal values, from the package `typing.Literal`. This change will involve updates to the SDK compiler, the pipeline run server and the UI input display.
+
+## Motivation
+Valid pipeline input currently includes string, int, float, boolean, list and struct, as well as two custom types – TaskFinalStatus and TaskConfig. None of these choices allows a pipeline author to restrict input to one or more pre-determined options. When a pipeline run is executed with unintended inputs, the resulting failure unexpected behavior not only wastes time and resources but also creates a frustrating user experience.
```

Comment:
Another motivation I'd add is to simplify the user experience. The user does not need to rely on the field being documented in the pipeline docstring to know what values are available. The KFP UI also doesn't currently surface these docstrings (the ODH/RHOAI dashboard does) if I recall correctly.